plugins/utils/which-key.nix

```
@@ -1,29 +1,239 @@
+    "showHelp"
+    "showKeys"
+    "triggersBlackList"
+    "triggersNoWait"
+  ];
+
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
+        "which-key"
+      ];
+      settingsPath = basePluginPath ++ [ "settings" ];
+    in
+    [
+      (lib.mkRenamedOptionModule
+        (
+          basePluginPath
+          ++ [
+            "disable"
+            "buftypes"
+          ]
+        )
+        (
+          settingsPath
+          ++ [
+            "disable"
+            "bt"
+          ]
+        )
+      )
+      (lib.mkRenamedOptionModule
+        (
+          basePluginPath
+          ++ [
+            "disable"
+            "filetypes"
+          ]
+        )
+        (
+          settingsPath
+          ++ [
+            "disable"
+            "ft"
+          ]
+        )
+      )
+      (lib.mkRenamedOptionModule
+        (
+          basePluginPath
+          ++ [
+            "window"
+            "winblend"
+          ]
+        )
+        (
+          settingsPath
+          ++ [
+            "win"
+            "wo"
+            "winblend"
+          ]
+        )
+      )
+      (lib.mkRenamedOptionModule
+        (
+          basePluginPath
+          ++ [
+            "window"
+            "border"
+          ]
+        )
+        (
+          settingsPath
+          ++ [
+            "win"
+            "border"
+          ]
+        )
+      )
+      (lib.mkRemovedOptionModule (basePluginPath ++ [ "triggers" ]) ''
+        Please use `plugins.which-key.settings.triggers` instead.
+
+        See the plugin documentation for more details about the new option signature.
+      '')
+      (lib.mkRemovedOptionModule
+        (
+          basePluginPath
+          ++ [
+            "window"
+            "padding"
+          ]
+        )
+        ''
+          Please use `plugins.which-key.settings.win.padding` instead.
+
+          See the plugin documentation for more details about the new option.
+        ''
+      )
+      (lib.mkRemovedOptionModule
+        (
+          basePluginPath
+          ++ [
+            "window"
+            "margin"
+          ]
+        )
+        ''
+
+          Removed in v3 rewrite.
+        ''
+      )
+      (lib.mkRemovedOptionModule
+        (
+          basePluginPath
+          ++ [
+            "window"
+            "position"
+          ]
+        )
+        ''
+          Please use `plugins.which-key.settings.layout`, `plugins.which-key.settings.win.title_pos`, or `plugins.which-key.settings.win.footer_pos` instead.
+
+          See the plugin documentation for more details about the new options.
+        ''
+      )
+    ];
+
+  settingsOptions = {
+    preset = defaultNullOpts.mkEnumFirstDefault [
+      "classic"
+      "modern"
+      "helix"
+      false
+    ] "Preset style for WhichKey. Set to false to disable.";
+
+    delay = defaultNullOpts.mkInt' {
+      pluginDefault = ''
+        function(ctx)
+          return ctx.plugin and 0 or 200
+        end'';
```

Needs to be rawLua. Also, we generally prefer to have a newline to make the string syntax more readable, unless you specifically want to avoid the trailing linebreak?


```suggestion
      pluginDefault.__raw = ''
        function(ctx)
          return ctx.plugin and 0 or 200
        end
      '';
```